msm/src/lib.rs

```diff
@@ -59,19 +59,38 @@ pub type OpeningProof = PairingProof<BN254>;
 mod tests {
     use crate::{
         columns::Column,
+        constraint::MSMCircuitEnv,
         lookups::{Lookup, LookupTableIDs},
         proof::ProofInputs,
         prover::prove,
         verifier::verify,
-        BaseSponge, Fp, OpeningProof, ScalarSponge, BN254,
+        BN254G1Affine, BaseSponge, Ff1, Fp, OpeningProof, ScalarSponge, BN254, MSM_FFADD_N_COLUMNS,
     };
     use ark_ff::UniformRand;
     use kimchi::circuits::domains::EvaluationDomains;
     use poly_commitment::pairing_proof::PairingSRS;
+    use rand::{thread_rng, Rng};
 
     // Number of columns
     const N: usize = 10;
 
+    // Creates a test witness for a * b = c constraint.
```

Comment:
I suppose it will disappear in next PRs when we will have FFA and FFM implemented, or simply the serialization.

Reply:
Partially, yes. One still needs to decide on a /particular/ instance we're testing even though the circuit/witness builder env is in place. Maybe it can be `random_witness` in `serialization/test.rs` or similarly in `ffa/`.

Even more generally, I thought we could just have a specific easy circuit just for testing, and avoid using `ffa/` and ` serialization/` altogether here.